File: 上學期/w10/AnUpdate.c
#include <stdio.h>
#include <string.h>

char table[26][26];
char s[100005], key[100005];
int main()
{
    for(int i = 0; i < 26; i++)
        for(int j = 0; j < 26; j++)
            table[i][j] = (i+j) % 26;

    scanf("%s%s", s, key);
    int klen = strlen(key);
    int slen = strlen(s);
    for(int i = 0; i < klen; i++)
        key[i] = ((key[i] >= 'A' && key[i] <= 'Z') ? key[i] - 'A' : key[i] - 'a');


    int index = 0;
    for(int i = 0; i < slen; i++)
    {
        if(s[i] >= 'A' && s[i] <= 'Z')
            s[i] = table[s[i] - 'A'][(int)key[index]] + 'A';
        else
            s[i] = table[s[i] - 'a'][(int)key[index]] + 'a';
        index = (index+1)%klen;
    }
    printf("%s\n", s);
    return 0;
}

File: 下學期/W1/AnUpdate.py
table = []

for i in range(26):
    a = []
    for j in range(26):
        a.append((i+j) % 26)
    table.append(a)

s, keys = map(str, input().split())

klen = len(keys)
keyn = []
for key in keys:
    if key >= 'A' and key <= 'Z':
        keyn.append(ord(key) - ord('A'))
    else:
        keyn.append(ord(key) - ord('a'))

index = 0
ans = ""
for c in s:
    if c >= 'A' and c <= 'Z':
        ans += chr(table[ord(c) - ord('A')][keyn[index]] + ord('A'))
    else:
       ans += chr(table[ord(c) - ord('a')][keyn[index]] + ord('a'))
    index = (index+1)%klen


print(ans)
